app/react/Map/helper.js

```diff
@@ -41,105 +57,106 @@ const markersToStyleFormat = markers =>
 
 const noop = () => {};
 
-const getPropertyBasedMarkers = (template, entity, color) => {
+const extractMarkers = ({ entity, geolocationProps, metadata, color, type }) => {
   const markers = [];
-
-  const geolocationProps = template.properties.filter(p => p.type === 'geolocation');
-  const geolocationPropNames = geolocationProps.map(p => p.name);
-
-  if (geolocationProps.length && entity.metadata) {
-    Object.keys(entity.metadata).forEach(property => {
-      if (geolocationPropNames.includes(property) && entity.metadata[property]) {
-        const { label } = geolocationProps.find(p => p.name === property);
-        entity.metadata[property]
-          .filter(mo => mo.value)
-          .forEach(({ value: point }) => {
-            const { lat, lon } = point;
-            markers.push({
-              properties: { entity, color, info: point.label },
-              latitude: lat,
-              longitude: lon,
-              label,
+  const geolocationPropNames = geolocationProps.map(prop => prop.get('name'));
+
+  if (geolocationProps.size && metadata) {
```

Comment:
Please refactor the nested if and else into functions

Reply:
Done